// @gonetone/hoyowiki-api/dist/utils/api.js
"use strict";
/*
 * Copyright (c) 2020-2022 原神資訊站 Genshin Impact Info
 * https://genshininfo.reh.tw/
 *
 * Developed by 旋風之音 GoneTone
 * https://github.com/GoneTone
 *
 *                               _oo0oo_
 *                              o8888888o
 *                              88" . "88
 *                              (| -_- |)
 *                              0\  =  /0
 *                            ___/`---'\___
 *                          .' \\|     |# '.
 *                         / \\|||  :  |||# \
 *                        / _||||| -:- |||||- \
 *                       |   | \\\  -  #/ |   |
 *                       | \_|  ''\---/''  |_/ |
 *                       \  .-\__  '-'  ___/-. /
 *                     ___'. .'  /--.--\  `. .'___
 *                  ."" '<  `.___\_<|>_/___.' >' "".
 *                 | | :  `- \`.;`\ _ /`;.`/ - ` : | |
 *                 \  \ `_.   \_ __\ /__ _/   .-` /  /
 *             =====`-.____`.___ \_____/___.-`___.-'=====
 *                               `=---='
 *           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 *               佛祖保佑                       永無 BUG
 *
 * GitHub: https://github.com/GoneTone/node-hoyowiki-api
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSetLanguage = exports.setLanguage = exports.axiosInstance = void 0;
const tslib_1 = require("tslib");
const axios_1 = tslib_1.__importDefault(require("axios"));
require("./constants");
const CharacterFilters_1 = require("../filters/CharacterFilters");
const WeaponFilters_1 = require("../filters/WeaponFilters");
const ArtifactFilters_1 = require("../filters/ArtifactFilters");
const EnemyFilters_1 = require("../filters/EnemyFilters");
const MaterialFilters_1 = require("../filters/MaterialFilters");
const NpcFilters_1 = require("../filters/NpcFilters");
const AnimalFilters_1 = require("../filters/AnimalFilters");
const TutorialFilters_1 = require("../filters/TutorialFilters");
exports.axiosInstance = axios_1.default.create({
    baseURL: "https://sg-wiki-api.hoyolab.com/hoyowiki/wapi" /* DefaultApiOptions.API */,
    headers: {
        'User-Agent': "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.63 Safari/537.36" /* DefaultApiOptions.UserAgent */,
        referer: "https://wiki.hoyolab.com" /* DefaultApiOptions.Referer */
    }
});
exports.axiosInstance.defaults.headers.common['x-rpc-language'] = "en-us" /* Language.EnglishUS */;
/**
 * Set Language.
 *
 * @param {Language} language Language Code
 */
async function setLanguage(language) {
    exports.axiosInstance.defaults.headers.common['x-rpc-language'] = language;
    await CharacterFilters_1.CharacterFilters.setFilterIds();
    await WeaponFilters_1.WeaponFilters.setFilterIds();
    await ArtifactFilters_1.ArtifactFilters.setFilterIds();
    await EnemyFilters_1.EnemyFilters.setFilterIds();
    await MaterialFilters_1.MaterialFilters.setFilterIds();
    await NpcFilters_1.NpcFilters.setFilterIds();
    await AnimalFilters_1.AnimalFilters.setFilterIds();
    await TutorialFilters_1.TutorialFilters.setFilterIds();
}
exports.setLanguage = setLanguage;
/**
 * Get Set Language.
 */
function getSetLanguage() {
    return exports.axiosInstance.defaults.headers.common['x-rpc-language'];
}
exports.getSetLanguage = getSetLanguage;
