// @gonetone/hoyowiki-api/dist/utils/constants.js
"use strict";
/*
 * Copyright (c) 2020-2022 原神資訊站 Genshin Impact Info
 * https://genshininfo.reh.tw/
 *
 * Developed by 旋風之音 GoneTone
 * https://github.com/GoneTone
 *
 *                               _oo0oo_
 *                              o8888888o
 *                              88" . "88
 *                              (| -_- |)
 *                              0\  =  /0
 *                            ___/`---'\___
 *                          .' \\|     |# '.
 *                         / \\|||  :  |||# \
 *                        / _||||| -:- |||||- \
 *                       |   | \\\  -  #/ |   |
 *                       | \_|  ''\---/''  |_/ |
 *                       \  .-\__  '-'  ___/-. /
 *                     ___'. .'  /--.--\  `. .'___
 *                  ."" '<  `.___\_<|>_/___.' >' "".
 *                 | | :  `- \`.;`\ _ /`;.`/ - ` : | |
 *                 \  \ `_.   \_ __\ /__ _/   .-` /  /
 *             =====`-.____`.___ \_____/___.-`___.-'=====
 *                               `=---='
 *           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 *               佛祖保佑                       永無 BUG
 *
 * GitHub: https://github.com/GoneTone/node-hoyowiki-api
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Language = exports.EntryPageMenu = exports.DefaultApiOptions = void 0;
/* eslint-disable no-unused-vars */
var DefaultApiOptions;
(function (DefaultApiOptions) {
    DefaultApiOptions["API"] = "https://sg-wiki-api.hoyolab.com/hoyowiki/wapi";
    DefaultApiOptions["UserAgent"] = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.63 Safari/537.36";
    DefaultApiOptions["Referer"] = "https://wiki.hoyolab.com";
})(DefaultApiOptions = exports.DefaultApiOptions || (exports.DefaultApiOptions = {}));
var EntryPageMenu;
(function (EntryPageMenu) {
    EntryPageMenu["Character"] = "2";
    EntryPageMenu["Weapon"] = "4";
    EntryPageMenu["Artifact"] = "5";
    EntryPageMenu["Enemy"] = "7";
    EntryPageMenu["Material"] = "9";
    EntryPageMenu["Npc"] = "10";
    EntryPageMenu["Book"] = "12";
    EntryPageMenu["Tutorial"] = "14";
    EntryPageMenu["Animal"] = "15";
})(EntryPageMenu = exports.EntryPageMenu || (exports.EntryPageMenu = {}));
var Language;
(function (Language) {
    Language["ChineseCN"] = "zh-cn";
    Language["ChineseTW"] = "zh-tw";
    Language["German"] = "de-de";
    Language["EnglishUS"] = "en-us";
    Language["SpanishES"] = "es-es";
    Language["French"] = "fr-fr";
    Language["Indonesia"] = "id-id";
    Language["Japanese"] = "ja-jp";
    Language["Korean"] = "ko-kr";
    Language["Portuguese"] = "pt-pt";
    Language["Russian"] = "ru-ru";
    Language["Thai"] = "th-th";
    Language["Vietnamese"] = "vi-vn";
})(Language = exports.Language || (exports.Language = {}));
