// @gonetone/hoyowiki-api/dist/index.js
"use strict";
/*
 * Copyright (c) 2020-2022 原神資訊站 Genshin Impact Info
 * https://genshininfo.reh.tw/
 *
 * Developed by 旋風之音 GoneTone
 * https://github.com/GoneTone
 *
 *                               _oo0oo_
 *                              o8888888o
 *                              88" . "88
 *                              (| -_- |)
 *                              0\  =  /0
 *                            ___/`---'\___
 *                          .' \\|     |# '.
 *                         / \\|||  :  |||# \
 *                        / _||||| -:- |||||- \
 *                       |   | \\\  -  #/ |   |
 *                       | \_|  ''\---/''  |_/ |
 *                       \  .-\__  '-'  ___/-. /
 *                     ___'. .'  /--.--\  `. .'___
 *                  ."" '<  `.___\_<|>_/___.' >' "".
 *                 | | :  `- \`.;`\ _ /`;.`/ - ` : | |
 *                 \  \ `_.   \_ __\ /__ _/   .-` /  /
 *             =====`-.____`.___ \_____/___.-`___.-'=====
 *                               `=---='
 *           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 *               佛祖保佑                       永無 BUG
 *
 * GitHub: https://github.com/GoneTone/node-hoyowiki-api
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchAPIInterface = exports.MenuFiltersAPIInterface = exports.EntryPageDataTutorialAPIInterface = exports.EntryPageDataBookAPIInterface = exports.EntryPageDataAnimalAPIInterface = exports.EntryPageDataNpcAPIInterface = exports.EntryPageDataMaterialAPIInterface = exports.EntryPageDataEnemyAPIInterface = exports.EntryPageDataArtifactAPIInterface = exports.EntryPageDataWeaponAPIInterface = exports.EntryPageDataCharacterAPIInterface = exports.EntryPageListDataTutorialAPIInterface = exports.EntryPageListDataBookAPIInterface = exports.EntryPageListDataAnimalAPIInterface = exports.EntryPageListDataNpcAPIInterface = exports.EntryPageListDataMaterialAPIInterface = exports.EntryPageListDataEnemyAPIInterface = exports.EntryPageListDataArtifactAPIInterface = exports.EntryPageListDataWeaponAPIInterface = exports.EntryPageListDataCharacterAPIInterface = void 0;
const tslib_1 = require("tslib");
tslib_1.__exportStar(require("./entryPageList/Character"), exports);
tslib_1.__exportStar(require("./entryPageList/Weapon"), exports);
tslib_1.__exportStar(require("./entryPageList/Artifact"), exports);
tslib_1.__exportStar(require("./entryPageList/Enemy"), exports);
tslib_1.__exportStar(require("./entryPageList/Material"), exports);
tslib_1.__exportStar(require("./entryPageList/Npc"), exports);
tslib_1.__exportStar(require("./entryPageList/Animal"), exports);
tslib_1.__exportStar(require("./entryPageList/Book"), exports);
tslib_1.__exportStar(require("./entryPageList/Tutorial"), exports);
tslib_1.__exportStar(require("./entryPage/Entry"), exports);
tslib_1.__exportStar(require("./search/Search"), exports);
tslib_1.__exportStar(require("./filters/CharacterFilters"), exports);
tslib_1.__exportStar(require("./filters/WeaponFilters"), exports);
tslib_1.__exportStar(require("./filters/ArtifactFilters"), exports);
tslib_1.__exportStar(require("./filters/EnemyFilters"), exports);
tslib_1.__exportStar(require("./filters/MaterialFilters"), exports);
tslib_1.__exportStar(require("./filters/NpcFilters"), exports);
tslib_1.__exportStar(require("./filters/AnimalFilters"), exports);
tslib_1.__exportStar(require("./filters/TutorialFilters"), exports);
tslib_1.__exportStar(require("./utils/api"), exports);
tslib_1.__exportStar(require("./utils/constants"), exports);
tslib_1.__exportStar(require("./interfaces/EntryPageListDataInterface"), exports);
exports.EntryPageListDataCharacterAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageListDataCharacterAPIInterface"));
exports.EntryPageListDataWeaponAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageListDataWeaponAPIInterface"));
exports.EntryPageListDataArtifactAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageListDataArtifactAPIInterface"));
exports.EntryPageListDataEnemyAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageListDataEnemyAPIInterface"));
exports.EntryPageListDataMaterialAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageListDataMaterialAPIInterface"));
exports.EntryPageListDataNpcAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageListDataNpcAPIInterface"));
exports.EntryPageListDataAnimalAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageListDataAnimalAPIInterface"));
exports.EntryPageListDataBookAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageListDataBookAPIInterface"));
exports.EntryPageListDataTutorialAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageListDataTutorialAPIInterface"));
exports.EntryPageDataCharacterAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageDataCharacterAPIInterface"));
exports.EntryPageDataWeaponAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageDataWeaponAPIInterface"));
exports.EntryPageDataArtifactAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageDataArtifactAPIInterface"));
exports.EntryPageDataEnemyAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageDataEnemyAPIInterface"));
exports.EntryPageDataMaterialAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageDataMaterialAPIInterface"));
exports.EntryPageDataNpcAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageDataNpcAPIInterface"));
exports.EntryPageDataAnimalAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageDataAnimalAPIInterface"));
exports.EntryPageDataBookAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageDataBookAPIInterface"));
exports.EntryPageDataTutorialAPIInterface = tslib_1.__importStar(require("./interfaces/EntryPageDataTutorialAPIInterface"));
exports.MenuFiltersAPIInterface = tslib_1.__importStar(require("./interfaces/MenuFiltersAPIInterface"));
exports.SearchAPIInterface = tslib_1.__importStar(require("./interfaces/SearchAPIInterface"));
tslib_1.__exportStar(require("./errors/HoYoWikiAPIError"), exports);
