// @gonetone/hoyowiki-api/dist/filters/AnimalFilters.js
"use strict";
/*
 * Copyright (c) 2020-2022 原神資訊站 Genshin Impact Info
 * https://genshininfo.reh.tw/
 *
 * Developed by 旋風之音 GoneTone
 * https://github.com/GoneTone
 *
 *                               _oo0oo_
 *                              o8888888o
 *                              88" . "88
 *                              (| -_- |)
 *                              0\  =  /0
 *                            ___/`---'\___
 *                          .' \\|     |# '.
 *                         / \\|||  :  |||# \
 *                        / _||||| -:- |||||- \
 *                       |   | \\\  -  #/ |   |
 *                       | \_|  ''\---/''  |_/ |
 *                       \  .-\__  '-'  ___/-. /
 *                     ___'. .'  /--.--\  `. .'___
 *                  ."" '<  `.___\_<|>_/___.' >' "".
 *                 | | :  `- \`.;`\ _ /`;.`/ - ` : | |
 *                 \  \ `_.   \_ __\ /__ _/   .-` /  /
 *             =====`-.____`.___ \_____/___.-`___.-'=====
 *                               `=---='
 *           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 *               佛祖保佑                       永無 BUG
 *
 * GitHub: https://github.com/GoneTone/node-hoyowiki-api
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnimalFilters = void 0;
const api_1 = require("../utils/api");
require("../utils/constants");
class AnimalFilters {
    static async setFilterIds() {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m;
        const response = await api_1.axiosInstance.get('/get_menu_filters', {
            params: {
                menu_id: "15" /* EntryPageMenu.Animal */
            }
        });
        const filters = response.data.data.filters;
        this.Type = {
            Other: (_c = (_b = (_a = filters[0]) === null || _a === void 0 ? void 0 : _a.values[0]) === null || _b === void 0 ? void 0 : _b.id) !== null && _c !== void 0 ? _c : '1458',
            Birds: (_f = (_e = (_d = filters[0]) === null || _d === void 0 ? void 0 : _d.values[1]) === null || _e === void 0 ? void 0 : _e.id) !== null && _f !== void 0 ? _f : '1471',
            Beasts: (_j = (_h = (_g = filters[0]) === null || _g === void 0 ? void 0 : _g.values[2]) === null || _h === void 0 ? void 0 : _h.id) !== null && _j !== void 0 ? _j : '1486',
            Fish: (_m = (_l = (_k = filters[0]) === null || _k === void 0 ? void 0 : _k.values[3]) === null || _l === void 0 ? void 0 : _l.id) !== null && _m !== void 0 ? _m : '1496'
        };
    }
}
exports.AnimalFilters = AnimalFilters;
Object.defineProperty(AnimalFilters, "Type", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: {
        Other: '1458',
        Birds: '1471',
        Beasts: '1486',
        Fish: '1496'
    }
});
