// @gonetone/hoyowiki-api/dist/filters/ArtifactFilters.js
"use strict";
/*
 * Copyright (c) 2020-2022 原神資訊站 Genshin Impact Info
 * https://genshininfo.reh.tw/
 *
 * Developed by 旋風之音 GoneTone
 * https://github.com/GoneTone
 *
 *                               _oo0oo_
 *                              o8888888o
 *                              88" . "88
 *                              (| -_- |)
 *                              0\  =  /0
 *                            ___/`---'\___
 *                          .' \\|     |# '.
 *                         / \\|||  :  |||# \
 *                        / _||||| -:- |||||- \
 *                       |   | \\\  -  #/ |   |
 *                       | \_|  ''\---/''  |_/ |
 *                       \  .-\__  '-'  ___/-. /
 *                     ___'. .'  /--.--\  `. .'___
 *                  ."" '<  `.___\_<|>_/___.' >' "".
 *                 | | :  `- \`.;`\ _ /`;.`/ - ` : | |
 *                 \  \ `_.   \_ __\ /__ _/   .-` /  /
 *             =====`-.____`.___ \_____/___.-`___.-'=====
 *                               `=---='
 *           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 *               佛祖保佑                       永無 BUG
 *
 * GitHub: https://github.com/GoneTone/node-hoyowiki-api
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArtifactFilters = void 0;
const api_1 = require("../utils/api");
require("../utils/constants");
class ArtifactFilters {
    static async setFilterIds() {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r, _s, _t, _u, _v, _w, _x, _y, _z, _0, _1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11;
        const response = await api_1.axiosInstance.get('/get_menu_filters', {
            params: {
                menu_id: "5" /* EntryPageMenu.Artifact */
            }
        });
        const filters = response.data.data.filters;
        this.ReliquaryEffect = {
            ATK: (_c = (_b = (_a = filters[0]) === null || _a === void 0 ? void 0 : _a.values[0]) === null || _b === void 0 ? void 0 : _b.id) !== null && _c !== void 0 ? _c : '240',
            DMGBonus: (_f = (_e = (_d = filters[0]) === null || _d === void 0 ? void 0 : _d.values[1]) === null || _e === void 0 ? void 0 : _e.id) !== null && _f !== void 0 ? _f : '254',
            HP: (_j = (_h = (_g = filters[0]) === null || _g === void 0 ? void 0 : _g.values[2]) === null || _h === void 0 ? void 0 : _h.id) !== null && _j !== void 0 ? _j : '265',
            ShieldStrength: (_m = (_l = (_k = filters[0]) === null || _k === void 0 ? void 0 : _k.values[3]) === null || _l === void 0 ? void 0 : _l.id) !== null && _m !== void 0 ? _m : '283',
            DEF: (_q = (_p = (_o = filters[0]) === null || _o === void 0 ? void 0 : _o.values[4]) === null || _p === void 0 ? void 0 : _p.id) !== null && _q !== void 0 ? _q : '294',
            CRITRate: (_t = (_s = (_r = filters[0]) === null || _r === void 0 ? void 0 : _r.values[5]) === null || _s === void 0 ? void 0 : _s.id) !== null && _t !== void 0 ? _t : '308',
            ElementalMastery: (_w = (_v = (_u = filters[0]) === null || _u === void 0 ? void 0 : _u.values[6]) === null || _v === void 0 ? void 0 : _v.id) !== null && _w !== void 0 ? _w : '319',
            EnergyRecharge: (_z = (_y = (_x = filters[0]) === null || _x === void 0 ? void 0 : _x.values[7]) === null || _y === void 0 ? void 0 : _y.id) !== null && _z !== void 0 ? _z : '333',
            CDReduction: (_2 = (_1 = (_0 = filters[0]) === null || _0 === void 0 ? void 0 : _0.values[8]) === null || _1 === void 0 ? void 0 : _1.id) !== null && _2 !== void 0 ? _2 : '346',
            HealingBonus: (_5 = (_4 = (_3 = filters[0]) === null || _3 === void 0 ? void 0 : _3.values[9]) === null || _4 === void 0 ? void 0 : _4.id) !== null && _5 !== void 0 ? _5 : '359',
            ElementalEffects: (_8 = (_7 = (_6 = filters[0]) === null || _6 === void 0 ? void 0 : _6.values[10]) === null || _7 === void 0 ? void 0 : _7.id) !== null && _8 !== void 0 ? _8 : '373',
            ElementalRES: (_11 = (_10 = (_9 = filters[0]) === null || _9 === void 0 ? void 0 : _9.values[11]) === null || _10 === void 0 ? void 0 : _10.id) !== null && _11 !== void 0 ? _11 : '384'
        };
    }
}
exports.ArtifactFilters = ArtifactFilters;
Object.defineProperty(ArtifactFilters, "ReliquaryEffect", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: {
        ATK: '240',
        DMGBonus: '254',
        HP: '265',
        ShieldStrength: '283',
        DEF: '294',
        CRITRate: '308',
        ElementalMastery: '319',
        EnergyRecharge: '333',
        CDReduction: '346',
        HealingBonus: '359',
        ElementalEffects: '373',
        ElementalRES: '384'
    }
});
