// @gonetone/hoyowiki-api/dist/filters/MaterialFilters.js
"use strict";
/*
 * Copyright (c) 2020-2022 原神資訊站 Genshin Impact Info
 * https://genshininfo.reh.tw/
 *
 * Developed by 旋風之音 GoneTone
 * https://github.com/GoneTone
 *
 *                               _oo0oo_
 *                              o8888888o
 *                              88" . "88
 *                              (| -_- |)
 *                              0\  =  /0
 *                            ___/`---'\___
 *                          .' \\|     |# '.
 *                         / \\|||  :  |||# \
 *                        / _||||| -:- |||||- \
 *                       |   | \\\  -  #/ |   |
 *                       | \_|  ''\---/''  |_/ |
 *                       \  .-\__  '-'  ___/-. /
 *                     ___'. .'  /--.--\  `. .'___
 *                  ."" '<  `.___\_<|>_/___.' >' "".
 *                 | | :  `- \`.;`\ _ /`;.`/ - ` : | |
 *                 \  \ `_.   \_ __\ /__ _/   .-` /  /
 *             =====`-.____`.___ \_____/___.-`___.-'=====
 *                               `=---='
 *           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 *               佛祖保佑                       永無 BUG
 *
 * GitHub: https://github.com/GoneTone/node-hoyowiki-api
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaterialFilters = void 0;
const api_1 = require("../utils/api");
require("../utils/constants");
class MaterialFilters {
    static async setFilterIds() {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r, _s, _t, _u, _v, _w, _x, _y, _z, _0, _1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, _16, _17, _18, _19, _20, _21, _22, _23, _24, _25, _26, _27, _28, _29, _30, _31, _32, _33, _34, _35, _36, _37, _38, _39, _40, _41, _42, _43, _44, _45, _46, _47, _48, _49, _50, _51, _52, _53, _54, _55, _56, _57, _58, _59, _60, _61, _62, _63, _64, _65, _66, _67, _68, _69, _70, _71, _72, _73, _74;
        const response = await api_1.axiosInstance.get('/get_menu_filters', {
            params: {
                menu_id: "9" /* EntryPageMenu.Material */
            }
        });
        const filters = response.data.data.filters;
        this.Type = {
            Material: (_c = (_b = (_a = filters[0]) === null || _a === void 0 ? void 0 : _a.values[0]) === null || _b === void 0 ? void 0 : _b.id) !== null && _c !== void 0 ? _c : '824',
            Consumable: (_f = (_e = (_d = filters[0]) === null || _d === void 0 ? void 0 : _d.values[1]) === null || _e === void 0 ? void 0 : _e.id) !== null && _f !== void 0 ? _f : '841',
            Potion: (_j = (_h = (_g = filters[0]) === null || _g === void 0 ? void 0 : _g.values[2]) === null || _h === void 0 ? void 0 : _h.id) !== null && _j !== void 0 ? _j : '853',
            SuperiorVoucher: (_m = (_l = (_k = filters[0]) === null || _k === void 0 ? void 0 : _k.values[3]) === null || _l === void 0 ? void 0 : _l.id) !== null && _m !== void 0 ? _m : '861',
            Bait: (_q = (_p = (_o = filters[0]) === null || _o === void 0 ? void 0 : _o.values[4]) === null || _p === void 0 ? void 0 : _p.id) !== null && _q !== void 0 ? _q : '877',
            LanternRiteMaterial: (_t = (_s = (_r = filters[0]) === null || _r === void 0 ? void 0 : _r.values[5]) === null || _s === void 0 ? void 0 : _s.id) !== null && _t !== void 0 ? _t : '892',
            WeaponEnhancementMaterial: (_w = (_v = (_u = filters[0]) === null || _u === void 0 ? void 0 : _u.values[6]) === null || _v === void 0 ? void 0 : _v.id) !== null && _w !== void 0 ? _w : '898',
            Food: (_z = (_y = (_x = filters[0]) === null || _x === void 0 ? void 0 : _x.values[7]) === null || _y === void 0 ? void 0 : _y.id) !== null && _z !== void 0 ? _z : '913',
            WeaponAscensionMaterial: (_2 = (_1 = (_0 = filters[0]) === null || _0 === void 0 ? void 0 : _0.values[8]) === null || _1 === void 0 ? void 0 : _1.id) !== null && _2 !== void 0 ? _2 : '924',
            ForgingBlueprint: (_5 = (_4 = (_3 = filters[0]) === null || _3 === void 0 ? void 0 : _3.values[9]) === null || _4 === void 0 ? void 0 : _4.id) !== null && _5 !== void 0 ? _5 : '948',
            CityStatesSigil: (_8 = (_7 = (_6 = filters[0]) === null || _6 === void 0 ? void 0 : _6.values[10]) === null || _7 === void 0 ? void 0 : _7.id) !== null && _8 !== void 0 ? _8 : '957',
            CommonVoucher: (_11 = (_10 = (_9 = filters[0]) === null || _9 === void 0 ? void 0 : _9.values[11]) === null || _10 === void 0 ? void 0 : _10.id) !== null && _11 !== void 0 ? _11 : '974',
            DragonspineMaterial: (_14 = (_13 = (_12 = filters[0]) === null || _12 === void 0 ? void 0 : _12.values[12]) === null || _13 === void 0 ? void 0 : _13.id) !== null && _14 !== void 0 ? _14 : '980',
            LocalSpecialtyLiyue: (_17 = (_16 = (_15 = filters[0]) === null || _15 === void 0 ? void 0 : _15.values[13]) === null || _16 === void 0 ? void 0 : _16.id) !== null && _17 !== void 0 ? _17 : '994',
            FishingRod: (_20 = (_19 = (_18 = filters[0]) === null || _18 === void 0 ? void 0 : _18.values[14]) === null || _19 === void 0 ? void 0 : _19.id) !== null && _20 !== void 0 ? _20 : '1013',
            QuestItem: (_23 = (_22 = (_21 = filters[0]) === null || _21 === void 0 ? void 0 : _21.values[15]) === null || _22 === void 0 ? void 0 : _22.id) !== null && _23 !== void 0 ? _23 : '1038',
            Gadget: (_26 = (_25 = (_24 = filters[0]) === null || _24 === void 0 ? void 0 : _24.values[16]) === null || _25 === void 0 ? void 0 : _25.id) !== null && _26 !== void 0 ? _26 : '1043',
            LimitedWishingItem: (_29 = (_28 = (_27 = filters[0]) === null || _27 === void 0 ? void 0 : _27.values[17]) === null || _28 === void 0 ? void 0 : _28.id) !== null && _29 !== void 0 ? _29 : '1054',
            CraftingBlueprint: (_32 = (_31 = (_30 = filters[0]) === null || _30 === void 0 ? void 0 : _30.values[18]) === null || _31 === void 0 ? void 0 : _31.id) !== null && _32 !== void 0 ? _32 : '1073',
            CookingIngredient: (_35 = (_34 = (_33 = filters[0]) === null || _33 === void 0 ? void 0 : _33.values[19]) === null || _34 === void 0 ? void 0 : _34.id) !== null && _35 !== void 0 ? _35 : '1085',
            RefinementMaterial: (_38 = (_37 = (_36 = filters[0]) === null || _36 === void 0 ? void 0 : _36.values[20]) === null || _37 === void 0 ? void 0 : _37.id) !== null && _38 !== void 0 ? _38 : '1093',
            CharacterEXPMaterial: (_41 = (_40 = (_39 = filters[0]) === null || _39 === void 0 ? void 0 : _39.values[21]) === null || _40 === void 0 ? void 0 : _40.id) !== null && _41 !== void 0 ? _41 : '1106',
            LocalSpecialtyInazuma: (_44 = (_43 = (_42 = filters[0]) === null || _42 === void 0 ? void 0 : _42.values[22]) === null || _43 === void 0 ? void 0 : _43.id) !== null && _44 !== void 0 ? _44 : '1141',
            CommonCurrency: (_47 = (_46 = (_45 = filters[0]) === null || _45 === void 0 ? void 0 : _45.values[23]) === null || _46 === void 0 ? void 0 : _46.id) !== null && _47 !== void 0 ? _47 : '1153',
            ArtifactEnhancementMaterial: (_50 = (_49 = (_48 = filters[0]) === null || _48 === void 0 ? void 0 : _48.values[24]) === null || _49 === void 0 ? void 0 : _49.id) !== null && _50 !== void 0 ? _50 : '1174',
            FurnishingBlueprint: (_53 = (_52 = (_51 = filters[0]) === null || _51 === void 0 ? void 0 : _51.values[25]) === null || _52 === void 0 ? void 0 : _52.id) !== null && _53 !== void 0 ? _53 : '1195',
            SpecialCurrency: (_56 = (_55 = (_54 = filters[0]) === null || _54 === void 0 ? void 0 : _54.values[26]) === null || _55 === void 0 ? void 0 : _55.id) !== null && _56 !== void 0 ? _56 : '1203',
            ForgingOre: (_59 = (_58 = (_57 = filters[0]) === null || _57 === void 0 ? void 0 : _57.values[27]) === null || _58 === void 0 ? void 0 : _58.id) !== null && _59 !== void 0 ? _59 : '1222',
            TalentLevelUpMaterial: (_62 = (_61 = (_60 = filters[0]) === null || _60 === void 0 ? void 0 : _60.values[28]) === null || _61 === void 0 ? void 0 : _61.id) !== null && _62 !== void 0 ? _62 : '1223',
            CharacterLevelUpMaterial: (_65 = (_64 = (_63 = filters[0]) === null || _63 === void 0 ? void 0 : _63.values[29]) === null || _64 === void 0 ? void 0 : _64.id) !== null && _65 !== void 0 ? _65 : '1246',
            LocalSpecialtyMondstadt: (_68 = (_67 = (_66 = filters[0]) === null || _66 === void 0 ? void 0 : _66.values[30]) === null || _67 === void 0 ? void 0 : _67.id) !== null && _68 !== void 0 ? _68 : '1263',
            AdventureItem: (_71 = (_70 = (_69 = filters[0]) === null || _69 === void 0 ? void 0 : _69.values[31]) === null || _70 === void 0 ? void 0 : _70.id) !== null && _71 !== void 0 ? _71 : '1285',
            Recipe: (_74 = (_73 = (_72 = filters[0]) === null || _72 === void 0 ? void 0 : _72.values[32]) === null || _73 === void 0 ? void 0 : _73.id) !== null && _74 !== void 0 ? _74 : '1297'
        };
    }
}
exports.MaterialFilters = MaterialFilters;
Object.defineProperty(MaterialFilters, "Type", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: {
        Material: '824',
        Consumable: '841',
        Potion: '853',
        SuperiorVoucher: '861',
        Bait: '877',
        LanternRiteMaterial: '892',
        WeaponEnhancementMaterial: '898',
        Food: '913',
        WeaponAscensionMaterial: '924',
        ForgingBlueprint: '948',
        CityStatesSigil: '957',
        CommonVoucher: '974',
        DragonspineMaterial: '980',
        LocalSpecialtyLiyue: '994',
        FishingRod: '1013',
        QuestItem: '1038',
        Gadget: '1043',
        LimitedWishingItem: '1054',
        CraftingBlueprint: '1073',
        CookingIngredient: '1085',
        RefinementMaterial: '1093',
        CharacterEXPMaterial: '1106',
        LocalSpecialtyInazuma: '1141',
        CommonCurrency: '1153',
        ArtifactEnhancementMaterial: '1174',
        FurnishingBlueprint: '1195',
        SpecialCurrency: '1203',
        ForgingOre: '1222',
        TalentLevelUpMaterial: '1223',
        CharacterLevelUpMaterial: '1246',
        LocalSpecialtyMondstadt: '1263',
        AdventureItem: '1285',
        Recipe: '1297'
    }
});
