// @gonetone/hoyowiki-api/dist/filters/WeaponFilters.js
"use strict";
/*
 * Copyright (c) 2020-2022 原神資訊站 Genshin Impact Info
 * https://genshininfo.reh.tw/
 *
 * Developed by 旋風之音 GoneTone
 * https://github.com/GoneTone
 *
 *                               _oo0oo_
 *                              o8888888o
 *                              88" . "88
 *                              (| -_- |)
 *                              0\  =  /0
 *                            ___/`---'\___
 *                          .' \\|     |# '.
 *                         / \\|||  :  |||# \
 *                        / _||||| -:- |||||- \
 *                       |   | \\\  -  #/ |   |
 *                       | \_|  ''\---/''  |_/ |
 *                       \  .-\__  '-'  ___/-. /
 *                     ___'. .'  /--.--\  `. .'___
 *                  ."" '<  `.___\_<|>_/___.' >' "".
 *                 | | :  `- \`.;`\ _ /`;.`/ - ` : | |
 *                 \  \ `_.   \_ __\ /__ _/   .-` /  /
 *             =====`-.____`.___ \_____/___.-`___.-'=====
 *                               `=---='
 *           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 *               佛祖保佑                       永無 BUG
 *
 * GitHub: https://github.com/GoneTone/node-hoyowiki-api
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeaponFilters = void 0;
const api_1 = require("../utils/api");
require("../utils/constants");
class WeaponFilters {
    static async setFilterIds() {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r, _s, _t, _u, _v, _w, _x, _y, _z, _0, _1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, _16, _17, _18, _19, _20, _21, _22, _23, _24, _25, _26, _27, _28, _29;
        const response = await api_1.axiosInstance.get('/get_menu_filters', {
            params: {
                menu_id: "4" /* EntryPageMenu.Weapon */
            }
        });
        const filters = response.data.data.filters;
        this.SecondaryAttributes = {
            ATKPercentage: (_c = (_b = (_a = filters[0]) === null || _a === void 0 ? void 0 : _a.values[0]) === null || _b === void 0 ? void 0 : _b.id) !== null && _c !== void 0 ? _c : '136',
            PhysicalDMGBonus: (_f = (_e = (_d = filters[0]) === null || _d === void 0 ? void 0 : _d.values[1]) === null || _e === void 0 ? void 0 : _e.id) !== null && _f !== void 0 ? _f : '151',
            DEFPercentage: (_j = (_h = (_g = filters[0]) === null || _g === void 0 ? void 0 : _g.values[2]) === null || _h === void 0 ? void 0 : _h.id) !== null && _j !== void 0 ? _j : '164',
            CRITRate: (_m = (_l = (_k = filters[0]) === null || _k === void 0 ? void 0 : _k.values[3]) === null || _l === void 0 ? void 0 : _l.id) !== null && _m !== void 0 ? _m : '176',
            CRITDMG: (_q = (_p = (_o = filters[0]) === null || _o === void 0 ? void 0 : _o.values[4]) === null || _p === void 0 ? void 0 : _p.id) !== null && _q !== void 0 ? _q : '190',
            ElementalMastery: (_t = (_s = (_r = filters[0]) === null || _r === void 0 ? void 0 : _r.values[5]) === null || _s === void 0 ? void 0 : _s.id) !== null && _t !== void 0 ? _t : '199',
            EnergyRecharge: (_w = (_v = (_u = filters[0]) === null || _u === void 0 ? void 0 : _u.values[6]) === null || _v === void 0 ? void 0 : _v.id) !== null && _w !== void 0 ? _w : '211',
            HPPercentage: (_z = (_y = (_x = filters[0]) === null || _x === void 0 ? void 0 : _x.values[7]) === null || _y === void 0 ? void 0 : _y.id) !== null && _z !== void 0 ? _z : '228'
        };
        this.Type = {
            Sword: (_2 = (_1 = (_0 = filters[1]) === null || _0 === void 0 ? void 0 : _0.values[0]) === null || _1 === void 0 ? void 0 : _1.id) !== null && _2 !== void 0 ? _2 : '398',
            Claymore: (_5 = (_4 = (_3 = filters[1]) === null || _3 === void 0 ? void 0 : _3.values[1]) === null || _4 === void 0 ? void 0 : _4.id) !== null && _5 !== void 0 ? _5 : '407',
            Bow: (_8 = (_7 = (_6 = filters[1]) === null || _6 === void 0 ? void 0 : _6.values[2]) === null || _7 === void 0 ? void 0 : _7.id) !== null && _8 !== void 0 ? _8 : '419',
            Catalyst: (_11 = (_10 = (_9 = filters[1]) === null || _9 === void 0 ? void 0 : _9.values[3]) === null || _10 === void 0 ? void 0 : _10.id) !== null && _11 !== void 0 ? _11 : '439',
            Polearm: (_14 = (_13 = (_12 = filters[1]) === null || _12 === void 0 ? void 0 : _12.values[4]) === null || _13 === void 0 ? void 0 : _13.id) !== null && _14 !== void 0 ? _14 : '443'
        };
        this.Quality = {
            Star5: (_17 = (_16 = (_15 = filters[2]) === null || _15 === void 0 ? void 0 : _15.values[0]) === null || _16 === void 0 ? void 0 : _16.id) !== null && _17 !== void 0 ? _17 : '673',
            Star4: (_20 = (_19 = (_18 = filters[2]) === null || _18 === void 0 ? void 0 : _18.values[1]) === null || _19 === void 0 ? void 0 : _19.id) !== null && _20 !== void 0 ? _20 : '683',
            Star3: (_23 = (_22 = (_21 = filters[2]) === null || _21 === void 0 ? void 0 : _21.values[2]) === null || _22 === void 0 ? void 0 : _22.id) !== null && _23 !== void 0 ? _23 : '696',
            Star2: (_26 = (_25 = (_24 = filters[2]) === null || _24 === void 0 ? void 0 : _24.values[3]) === null || _25 === void 0 ? void 0 : _25.id) !== null && _26 !== void 0 ? _26 : '712',
            Star1: (_29 = (_28 = (_27 = filters[2]) === null || _27 === void 0 ? void 0 : _27.values[4]) === null || _28 === void 0 ? void 0 : _28.id) !== null && _29 !== void 0 ? _29 : '724'
        };
    }
}
exports.WeaponFilters = WeaponFilters;
Object.defineProperty(WeaponFilters, "SecondaryAttributes", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: {
        ATKPercentage: '136',
        PhysicalDMGBonus: '151',
        DEFPercentage: '164',
        CRITRate: '176',
        CRITDMG: '190',
        ElementalMastery: '199',
        EnergyRecharge: '211',
        HPPercentage: '228'
    }
});
Object.defineProperty(WeaponFilters, "Type", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: {
        Sword: '398',
        Claymore: '407',
        Bow: '419',
        Catalyst: '439',
        Polearm: '443'
    }
});
Object.defineProperty(WeaponFilters, "Quality", {
    enumerable: true,
    configurable: true,
    writable: true,
    value: {
        Star5: '673',
        Star4: '683',
        Star3: '696',
        Star2: '712',
        Star1: '724'
    }
});
